### PR TITLE
Add: Transaction Id to Request-Response Headers

### PR DESCRIPTION
**Pull request by keyur9**
Add: Transaction Id to Request-Response Headers

### CODE REVIEW

File: moesifmiddleware.go [hunk 1]
```diff
@@ -75,6 +88,33 @@
...
+		disableTransactionId := false
+		// Try to fetch the disableTransactionId from the option
+		if isEnabled, found := moesifOption["disableTransactionId"].(bool); found {
+			disableTransactionId = isEnabled
+		}
+
+		// Add transactionId to the headers
+		if !disableTransactionId {
+			// Try to fetch the transactionId from the header
+			reqTransId := request.Header.Get("X-Moesif-Transaction-Id")
```
Comment (matthewoates) [inline moesifmiddleware.go]: why not `transactionId = request.Header.Get("X-Moesif-Transaction-Id")`

File: moesifmiddleware.go [hunk 2]
```diff
@@ -75,6 +88,33 @@
...
+			// Set the transationId
+			transationId := reqTransId
+			// Check if need to generate transactionId
+			if len(transationId) == 0 {
```
Comment (matthewoates) [inline moesifmiddleware.go]: Is it possible for `transactionId` to be `nil`?

File: moesifmiddleware.go (hunk 1 above)
Comment (keyur9) [reply moesifmiddleware.go]: have refactored the code

File: moesifmiddleware.go (hunk 2 above)
Comment (keyur9) [reply moesifmiddleware.go]: yes, it is possible for transactionId to be nil if the user has set the flag

**Pull request merged**